uxarray/grid.py

```diff
@@ -380,6 +380,48 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
 
         return integral
 
+    def _build_edge_node_connectivity(self):
+
+        # padded face nodes: [nMesh2_face x nMaxMesh2_face_nodes + 1]
+        padded_face_nodes = _close_face_nodes(self.Mesh2_face_nodes.values,
+                                              self.nMesh2_face,
+                                              self.nMaxMesh2_face_nodes)
+
+        # construct an array of empty edge nodes where each entry is a pair of indices
+        edge_nodes = np.empty((self.nMesh2_face * self.nMaxMesh2_face_nodes, 2),
+                              dtype=INT_DTYPE)
+
+        # first index includes starting node up to non-padded value
+        edge_nodes[:, 0] = padded_face_nodes[:, :-1].ravel()
+
+        # second index includes second node up to padded value
+        edge_nodes[:, 1] = padded_face_nodes[:, 1:].ravel()
+
+        # all edge nodes that contain a fill value
+        fill_value_mask = np.logical_or(edge_nodes[:, 0] == INT_FILL_VALUE,
+                                        edge_nodes[:, 1] == INT_FILL_VALUE)
+
+        # all edge nodes that do not contain a fill value
+        non_fill_value_mask = np.logical_not(fill_value_mask)
+
+        # filter out all invalid edges
+        edge_nodes = edge_nodes[non_fill_value_mask]
```

Comment:
@hongyuchen1030

This is the only section that we still need to finalize. The question that remains is whether we want to filter out unique edges from here. Since entries in `Mesh2_edge_nodes` don't have any spatial relationship to each other, as long as we properly index into it using `Mesh2_face_edges`, it should not be an issue. What do you think?

Reply:
I am not very clear why we don't need to use unique edges here, Can you elaborate it a bit with some usage examples? Thanks

Reply:
Actually scrap my idea, I did a bit more digging into how `np.unique()` works, and we should be able to use the inverse indices to construct our `face_edges`. Will be pushing a few changes here shortly

Reply:
I think that's exactly what I did in the `build_face_edges_connectivity`, and also why I think building `build_face_edges_connectivity` and `build_edge_nodes_connectivity` together will be more efficient. But probably from the users' standpoints, having another `build_edge_nodes_connectivity` will be more intuitive.

Reply:
I prefer having them separate, since it's easier to debug and test. 

I also made the slight change to make these functions protected, since they will be called within the Grid object, not by the user.

Reply:
Works for me